File: wali-wallet/wallet-ui-interface/web-extension/convert-svg-to-png.js
#!/usr/bin/env node

/**
 * Convert SVG walrus logo to PNG icons
 * Uses Puppeteer to render SVG and screenshot at different sizes
 */

const fs = require('fs');
const path = require('path');

// Try to use puppeteer if available, otherwise use a simpler approach
let usePuppeteer = false;
try {
  require.resolve('puppeteer');
  usePuppeteer = true;
} catch (e) {
  console.log('Puppeteer not found, trying alternative method...');
}

if (usePuppeteer) {
  convertWithPuppeteer();
} else {
  convertWithSharp();
}

async function convertWithSharp() {
  try {
    const sharp = require('sharp');
    const svgPath = path.join(__dirname, '../assets/wali-logo.svg');
    const outputDir = path.join(__dirname, 'public/assets');
    
    // Ensure output directory exists
    if (!fs.existsSync(outputDir)) {
      fs.mkdirSync(outputDir, { recursive: true });
    }
    
    const svgBuffer = fs.readFileSync(svgPath);
    const sizes = [16, 48, 128];
    
    console.log('Converting SVG to PNG icons with sharp...');
    
    for (const size of sizes) {
      const outputPath = path.join(outputDir, `icon-${size}.png`);
      await sharp(svgBuffer)
        .resize(size, size)
        .png()
        .toFile(outputPath);
      console.log(`✓ Created ${size}x${size} icon: ${outputPath}`);
    }
    
    console.log('\n🦭 All walrus icons created successfully!');
  } catch (error) {
    console.error('Sharp method failed:', error.message);
    console.log('\nTrying browser-based conversion...');
    convertWithBrowser();
  }
}

async function convertWithPuppeteer() {
  const puppeteer = require('puppeteer');
  const svgPath = path.join(__dirname, '../assets/wali-logo.svg');
  const outputDir = path.join(__dirname, 'public/assets');
  
  // Ensure output directory exists
  if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
  }
  
  const svgContent = fs.readFileSync(svgPath, 'utf8');
  const sizes = [16, 48, 128];
  
  console.log('Converting SVG to PNG icons with Puppeteer...');
  
  const browser = await puppeteer.launch({ headless: true });
  const page = await browser.newPage();
  
  for (const size of sizes) {
    await page.setViewport({ width: size, height: size });
    await page.setContent(`
      <!DOCTYPE html>
      <html>
        <head>
          <style>
            body { margin: 0; padding: 0; width: ${size}px; height: ${size}px; }
            svg { width: 100%; height: 100%; }
          </style>
        </head>
        <body>${svgContent}</body>
      </html>
    `);
    
    const outputPath = path.join(outputDir, `icon-${size}.png`);
    await page.screenshot({ path: outputPath, omitBackground: false });
    console.log(`✓ Created ${size}x${size} icon: ${outputPath}`);
  }
  
  await browser.close();
  console.log('\n🦭 All walrus icons created successfully!');
}

function convertWithBrowser() {
  console.log('\n📝 Manual conversion needed:');
  console.log('Please run this HTML file in a browser and save the canvas images:\n');
  
  const svgPath = path.join(__dirname, '../assets/wali-logo.svg');
  const svgContent = fs.readFileSync(svgPath, 'utf8');
  
  const htmlContent = `
<!DOCTYPE html>
<html>
<head>
  <title>SVG to PNG Converter</title>
  <style>
    body { font-family: Arial, sans-serif; padding: 20px; }
    .icon-container { display: inline-block; margin: 20px; text-align: center; }
    canvas { border: 1px solid #ddd; display: block; margin: 10px auto; }
    button { padding: 10px 20px; margin: 5px; cursor: pointer; }
  </style>
</head>
<body>
  <h1>🦭 wAli Walrus Logo Converter</h1>
  <p>Click the download buttons to save each icon:</p>
  
  <div class="icon-container">
    <h3>16x16 (Toolbar)</h3>
    <canvas id="canvas-16" width="16" height="16"></canvas>
    <button onclick="download(16)">Download 16x16</button>
  </div>
  
  <div class="icon-container">
    <h3>48x48 (Extension)</h3>
    <canvas id="canvas-48" width="48" height="48"></canvas>
    <button onclick="download(48)">Download 48x48</button>
  </div>
  
  <div class="icon-container">
    <h3>128x128 (Store)</h3>
    <canvas id="canvas-128" width="128" height="128"></canvas>
    <button onclick="download(128)">Download 128x128</button>
  </div>
  
  <script>
    const svgContent = \`${svgContent}\`;
    const sizes = [16, 48, 128];
    
    function renderSVG(size) {
      const canvas = document.getElementById('canvas-' + size);
      const ctx = canvas.getContext('2d');
      const img = new Image();
      const blob = new Blob([svgContent], { type: 'image/svg+xml' });
      const url = URL.createObjectURL(blob);
      
      img.onload = function() {
        ctx.drawImage(img, 0, 0, size, size);
        URL.revokeObjectURL(url);
      };
      
      img.src = url;
    }
    
    function download(size) {
      const canvas = document.getElementById('canvas-' + size);
      canvas.toBlob(function(blob) {
        const url = URL.createObjectURL(blob);
        const a = document.createElement('a');
        a.href = url;
        a.download = 'icon-' + size + '.png';
        a.click();
        URL.revokeObjectURL(url);
      });
    }
    
    // Render all sizes on load
    sizes.forEach(size => renderSVG(size));
  </script>
</body>
</html>
`;
  
  const htmlPath = path.join(__dirname, 'convert-icons.html');
  fs.writeFileSync(htmlPath, htmlContent);
  console.log(`Created: ${htmlPath}`);
  console.log('Open this file in your browser and click the download buttons.');
}

File: wali-wallet/wallet-ui-interface/web-extension/convert_svg_to_png.py
#!/usr/bin/env python3
"""
Convert SVG walrus logo to PNG icons for Chrome extension
Uses cairosvg or Pillow + reportlab, with fallbacks
"""

import sys
import os
from pathlib import Path

def convert_with_cairosvg(svg_path, output_dir, sizes):
    """Use cairosvg (best quality)"""
    try:
        import cairosvg
        print("Using cairosvg for conversion...")
        
        for size in sizes:
            output_path = os.path.join(output_dir, f'icon-{size}.png')
            cairosvg.svg2png(
                url=svg_path,
                write_to=output_path,
                output_width=size,
                output_height=size
            )
            print(f"✓ Created {size}×{size} icon: {output_path}")
        return True
    except ImportError:
        print("cairosvg not available")
        return False
    except Exception as e:
        print(f"cairosvg failed: {e}")
        return False

def convert_with_pillow(svg_path, output_dir, sizes):
    """Use Pillow with svglib"""
    try:
        from PIL import Image
        from svglib.svglib import svg2rlg
        from reportlab.graphics import renderPM
        
        print("Using Pillow + svglib for conversion...")
        
        for size in sizes:
            # Convert SVG to ReportLab drawing
            drawing = svg2rlg(svg_path)
            
            # Scale the drawing
            scale = size / 200.0  # Original SVG is 200x200
            drawing.width = size
            drawing.height = size
            drawing.scale(scale, scale)
            
            # Render to PNG
            output_path = os.path.join(output_dir, f'icon-{size}.png')
            renderPM.drawToFile(drawing, output_path, fmt='PNG')
            print(f"✓ Created {size}×{size} icon: {output_path}")
        return True
    except ImportError as e:
        print(f"Pillow/svglib not available: {e}")
        return False
    except Exception as e:
        print(f"Pillow method failed: {e}")
        return False

def install_dependencies():
    """Try to install required dependencies"""
    import subprocess
    
    print("\nAttempting to install cairosvg...")
    try:
        subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'cairosvg'])
        return True
    except:
        print("Failed to install cairosvg")
        
    print("\nAttempting to install pillow svglib reportlab...")
    try:
        subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'pillow', 'svglib', 'reportlab'])
        return True
    except:
        print("Failed to install pillow dependencies")
        
    return False

def main():
    # Paths
    script_dir = Path(__file__).parent
    svg_path = script_dir.parent / 'assets' / 'wali-logo.svg'
    output_dir = script_dir / 'public' / 'assets'
    
    # Ensure paths exist
    if not svg_path.exists():
        print(f"Error: SVG file not found: {svg_path}")
        return 1
    
    output_dir.mkdir(parents=True, exist_ok=True)
    
    # Sizes to generate
    sizes = [16, 48, 128]
    
    print(f"Converting {svg_path} to PNG icons...")
    print(f"Output directory: {output_dir}\n")
    
    # Try cairosvg first (best quality)
    if convert_with_cairosvg(str(svg_path), str(output_dir), sizes):
        print("\n🦭 All walrus icons created successfully with cairosvg!")
        return 0
    
    # Try Pillow + svglib
    if convert_with_pillow(str(svg_path), str(output_dir), sizes):
        print("\n🦭 All walrus icons created successfully with Pillow!")
        return 0
    
    # If nothing worked, try installing dependencies
    print("\n❌ No SVG converters available.")
    response = input("Would you like to try installing dependencies? (y/n): ")
    
    if response.lower().startswith('y'):
        if install_dependencies():
            print("\nRetrying conversion...")
            if convert_with_cairosvg(str(svg_path), str(output_dir), sizes):
                print("\n🦭 All walrus icons created successfully!")
                return 0
            if convert_with_pillow(str(svg_path), str(output_dir), sizes):
                print("\n🦭 All walrus icons created successfully!")
                return 0
    
    print("\n❌ Conversion failed. Please install cairosvg or pillow+svglib manually:")
    print("  pip install cairosvg")
    print("  OR")
    print("  pip install pillow svglib reportlab")
    return 1

if __name__ == '__main__':
    sys.exit(main())
